src/client/App.jsx

```diff
@@ -0,0 +1,22 @@
+import { createContext } from "react";
+import { Route, Routes, Link } from "react-router-dom";
+import Dashboard from "./components/Dashboard";
+import DisplayUsers from "./components/DisplayUsers";
+
+export const appContext = createContext();
+
+const port = import.meta.env.VITE_PORT;
+const apiUrl = `http://localhost:${port}`;
```

Comment:
you could have just set this as a const in a separate file, like an `api.js` file that exports the API_URL variable, so there would have not been any need for a context for this